src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs

```diff
@@ -145,5 +195,101 @@ internal static bool TryCoerceTo(this FormulaValue value, FormattingInfo formatI
         {
             return TryGetDateTime(formatInfo, IRContext.NotInSource(FormulaType.DateTime), value, out result);
         }
+
+        /// <summary>
+        /// Try to convert value to Record format.
+        /// </summary>
+        /// <param name="value">Input value.</param>
+        /// <param name="targetType">Target type.</param>
+        /// <param name="result">Result value.</param>
+        /// <returns>True/False based on whether function can convert from original type to Record type.</returns> 
+        public static bool TryCoerceTo(this RecordValue value, RecordType targetType, out RecordValue result)
+        {
+            result = null;
+            var recordResult = new NamedValue[value.Fields.Count()];
```

Comment:
>value.Fields.Count() [](http://example.com/codeflow?start=46&length=20)

nit: 
if we're traversing parallel arrays. 

len = value.Fields.Count();

for(int i = 0; i < len; i++) 
{


}

Reply:
update here